### PR TITLE
[torchlib] Fix reflection pad

### PR DESCRIPTION
**Pull request by justinchuby**

### CODE REVIEW

File: onnxscript/function_libs/torch_lib/ops/nn.py [hunk 1]
```diff
@@ -1517,37 +1519,16 @@ def aten_reflection_pad1d_backward(
...
-    odd_elements = op.Slice(torch_paddings, starts, ends, zero, steps)
-    # paddings[-1::-2]
-    starts = neg_1
-    ends = op.Sub(starts, size_d)
-    even_elements = op.Slice(torch_paddings, starts, ends, zero, steps)
-    # paddings[-2::-2] + paddings[-1::-2]
-    onnx_padding = op.Concat(odd_elements, even_elements, axis=0)
+    rank = len(self.shape)
+    paddings = list(padding) + [0] * (rank * 2 - len(padding))
+    paddings = paddings[-2::-2] + paddings[-1::-2]
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/nn.py:1529]: Why reversed?

**Review by titaiwangms — APPROVED** [review]
Do we need a new reflection_pad test to guard the missing corner case?

**Comment by justinchuby** [comment]
> Do we need a new reflection_pad test to guard the missing corner case?

Could you remind me which was the missing case?

**Comment by titaiwangms** [comment]
> pytorch/pytorch#144382

Aren't we refactoring because it has this corner case?

**Comment by justinchuby** [comment]
Let me test

**Comment by justinchuby** [comment]
Thanks for the reminder. It's fixed. Would be helpful to create some cases. I will do that
